### DISCUSSION TITLE
Self hosting

### DISCUSSION THREAD

**Discussion by Fractal-Tess** [supabase/supabase#7967] [Questions]
Hey there.

The self-hosting section/documentation on the main website is extremely minimal and it doesn't go into great detail as to how to set up the services. Today I tried following the 4-line example with docker on this [page](https://supabase.com/docs/guides/hosting/docker) but with no success. The studio is querying the `localhost:8000/api/v1`, but that request never gets responded to.
![image](https://user-images.githubusercontent.com/75957529/182024757-589eabcc-e103-41ee-9834-7b7fc9027511.png)

Do I need to do some sort of database initialization? Is it a user error on my part? 
All I'm doing is following the basic example provided by superbase on their official doc.
```bash
# Get the code
git clone --depth 1 https://github.com/supabase/supabase

# Go to the docker folder
cd supabase/docker

# Copy the fake env vars
cp .env.example .env

# Start
docker-compose up
```

**Comment by vogelcodes** [comment]
Have you changed the secrets from .env.example? If you did, try changin in `./volumes/api/kong.yml`

**Reply by Fractal-Tess** [reply]
Didn't change anything. Left everything as bare bones as it comes.
My suspicion is there is something weird going on when the address is being resolved.
This is being run by a RP that is on my network, however, if I do run it from the same machine, I'm able to get full functionality. 
I explained it in detail on the superbase discord server, over [here](https://discord.com/channels/839993398554656828/1003255470338883614)
If I get a response there and solve my issue, I'll copy-paste it here, so others with similar issues can solve it as well.

**Reply by vogelcodes** [reply]
If you're running it in another machine, the supabase-studio calls to localhost:8000 wont work because the api is running in your RP and the studio is running in your browser. 

Try updating these variables to target your RP IP
##General
SITE_URL=
API_EXTERNAL_URL=

PUBLIC_REST_URL=

**Reply by vogelcodes** [reply]
I think we can help each other out a lot, add me on Discord: UrsoowW#5287

**Reply by Fractal-Tess** [reply]
> 

Done.

**Comment by Fractal-Tess** [comment]
Also, here is another example.
I span up a Linode instance (arch). And then did the following commands
```bash
pacman -Syu
pacman -S docker docker-compose git
systemctl start docker

#copy paste the supabase example
# Get the code
git clone --depth 1 https://github.com/supabase/supabase

# Go to the docker folder
cd supabase/docker

# Copy the fake env vars
cp .env.example .env

# Start
docker-compose up
```

Now the Linode server is running on 172.104.242.91.

Navigating to `http://172.104.242.91:3000` has basically the same issue.

**Comment by Fractal-Tess** [comment]
Update:
I tried narrow it down even more, and on the Linode server, I changed the PUBLIC_REST_URL endpoint for the rest service to 172.104242.91:8000/rest - which in turn made it work.
But doing the same thing on my raspberry pi doesn't yield the same result
I'm still getting the 502 error from kong
Closing the Linode server now. 
All that is left, i figuring out why the kong api is giving back a 502 status code.

**Comment by Fractal-Tess** [comment]
Turns out, the main problem is that the database container fails to run on arm. Well, to no surprise, it is marked as amd64 build, and the latest arm support is 4 months old - what is worse, it doesn't even run. 
```bash
sudo docker run --rm -it --platform=linux/arm64 -e POSTGRES_PASSSWORD=postgres supabase/postgres@sha256:1110e0cc0ad988355630c6c78c2bd3b1e761bb59e19bad4e9f28114c38580642
# -> exec /usr/local/bin/docker-entrypoint.sh: exec format error

sudo docker run --rm -it --platform=linux/arm64 -e POSTGRES_PASSSWORD=postgres supabase/postgres:14.1.0.15
# -> exec /usr/local/bin/docker-entrypoint.sh: exec format error
```

**Reply by jtkeyva** [reply]
@Fractal-Tess Any luck with this? I'm on M1 and it doesn't get past the connection screen

**Reply by jtkeyva** [reply]
Actually I just got https://supabase.com/docs/guides/cli/local-development working on my M1. I had to crank up the RAM in Docker to 16GB and now it runs fine.

**Reply by Fractal-Tess** [reply]
> Actually I just got https://supabase.com/docs/guides/cli/local-development working on my M1. I had to crank up the RAM in Docker to 16GB and now it runs fine.

For me, I had to build the postgres docker container from source. Also there is one that works on arm already on docker hub.

**Reply by jtkeyva** [reply]
Thanks! That arm one on docker hub is 5 months old, hopefully they keep them in sync.
